mlflow/xgboost.py

```diff
@@ -461,15 +461,16 @@ def log_feature_importance_plot(features, importance, importance_type):
                 # pylint: disable=undefined-loop-variable
                 filepath = os.path.join(tmpdir, "feature_importance_{}.png".format(imp_type))
                 fig.savefig(filepath)
-                try_mlflow_log(mlflow.log_artifact, filepath)
+                mlflow.log_artifact(filepath)
             finally:
                 plt.close(fig)
                 shutil.rmtree(tmpdir)
 
-        # logging booster params separately via mlflow.log_params to extract key/value pairs
-        # and make it easier to compare them across runs.
-        params = args[0] if len(args) > 0 else kwargs["params"]
-        try_mlflow_log(mlflow.log_params, params)
+        autologging_client = MlflowAutologgingQueueingClient()
```

Comment:
Not specific to this PR. But the train() function is getting really long and difficult to read. It might be good to factor some code out (e.g. move scoped functions to the top, and use with statement for both autologging_client) in a follow up cleanup PR.

Reply:
Agreed! I'll file a ticket to track cleaning this up